app/tests/test_keycloak_manager.py

```diff
@@ -0,0 +1,173 @@
+        )
+    assert len(user_groups_list) == 0
+
+
+@patch(
+    "app.main.authorize.access_token_sign_in_required.keycloak.KeycloakOpenID.introspect"
+)
+def test_get_user_transferring_body_groups_active_access_token_invalid_transferring_body(
+    mock_decode_keycloak_access_token, app
+):
+    """
+    Given an active access token in the session,
+    When access token given
+    Then it should not return user group list
+    """
+    mock_decode_keycloak_access_token.return_value = {
+        "active": True,
+        "groups": ["/application_1/foo", "/application_2/bar"],
+    }
+    app.config["FORCE_AUTHENTICATION_FOR_IN_TESTING"] = True
+    with app.test_client() as client:
+        with client.session_transaction() as session:
+            session["access_token"] = "valid_token"
+
+        for group in mock_decode_keycloak_access_token.return_value["groups"]:
+            index = group.find("transferring_body")
+
+        assert index == -1
+
+
+@patch(
+    "app.main.authorize.access_token_sign_in_required.keycloak.KeycloakOpenID.introspect"
+)
+def test_get_user_transferring_body_groups_multiple_transferring_body(
+    mock_decode_keycloak_access_token, app
+):
+    """
+    Given an active access token in the session,
+    When access token given
+    Then it should return 2 items in user group list
+    """
+    mock_decode_keycloak_access_token.return_value = {
+        "active": True,
+        "groups": ["/transferring_body/foo", "/transferring_body/bar"],
+    }
+    user_groups_list = []
+    app.config["FORCE_AUTHENTICATION_FOR_IN_TESTING"] = True
+    with app.test_client() as client:
+        with client.session_transaction() as session:
+            session["access_token"] = "valid_token"
+
+        for group in mock_decode_keycloak_access_token.return_value["groups"]:
+            index = group.find("transferring_body")
+            print(index)
+            if index != -1:
+                user_groups_list.append(group[2])
+
+        assert len(user_groups_list) == 2
+
+
+@patch(
+    "app.main.authorize.access_token_sign_in_required.keycloak.KeycloakOpenID.introspect"
+)
+def test_get_user_transferring_body_groups_return_multiple_transferring_body(
+    mock_decode_keycloak_access_token, app
+):
+    """
+    Given an active access token in the session,
+    When access token given
+    Then it should return 2 items in user group list
+    """
+    mock_decode_keycloak_access_token.return_value = {
+        "active": True,
+        "groups": [
+            "/transferring_body/foo",
+            "/transferring_body/bar",
+            "/ayr_user/bar",
+        ],
+    }
+    user_groups_list = []
+    app.config["FORCE_AUTHENTICATION_FOR_IN_TESTING"] = True
+    with app.test_client() as client:
+        with client.session_transaction() as session:
+            session["access_token"] = "valid_token"
+
+        for group in mock_decode_keycloak_access_token.return_value["groups"]:
```

Comment:
This test is still not testing what we want to test. It is building up a list of length 2 and showing that the length is 2.